ostd/src/cpu/mod.rs

```diff
@@ -21,6 +21,70 @@ use crate::{
     arch::boot::smp::get_num_processors, task::DisabledPreemptGuard, trap::DisabledLocalIrqGuard,
 };
 
+/// The ID of a CPU in the system.
+///
+/// If converting from/to an integer, the integer must start from 0 and be less
+/// than the number of CPUs.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CpuId(u32);
+
+/// An error indicating that the CPU does not exist.
+#[derive(Debug)]
+pub struct NonexistentCPU(pub usize);
```

Comment:
I can't see the meaning of introduing  `NonexistentCPU`. Maybe you need a new Error type like `Error::NonexistentCPU` instead of a new type?